CMakeLists.txt

```diff
@@ -241,15 +243,28 @@ if (HDF5)
 endif()  
 
 # Parmetis
-if (METIS)
+if ("metis" IN_LIST GRAPH_PARTITIONING_LIBS)
   find_package(METIS REQUIRED)
   if (NOT METIS_64_BIT_INTEGER)
     message(WARNING "Found METIS which does not support IDXTYPEWIDTH = 64. But this is required for SeisSol.")
   endif()
   find_package(ParMETIS REQUIRED)
   target_include_directories(SeisSol-common-properties INTERFACE ${PARMETIS_INCLUDE_DIRS})
   target_link_libraries(SeisSol-common-properties INTERFACE ${PARMETIS_LIBRARIES})
-  target_compile_definitions(SeisSol-common-properties INTERFACE USE_METIS)
+  target_compile_definitions(SeisSol-common-properties INTERFACE USE_METIS USE_PARMETIS HAVE_GRAPH_PARTITIONING_LIBRARY)
+endif()
+if ("parhip" IN_LIST GRAPH_PARTITIONING_LIBS)
+    find_package(ParHIP REQUIRED)
+    target_link_libraries(SeisSol-common-properties INTERFACE ParHIP::ParHIP)
+    target_compile_definitions(SeisSol-common-properties INTERFACE USE_PARHIP HAVE_GRAPH_PARTITIONING_LIBRARY)
+endif()
+if ("scotch" IN_LIST GRAPH_PARTITIONING_LIBS)
+    find_package(SCOTCH REQUIRED)
+    target_link_libraries(SeisSol-common-properties INTERFACE SCOTCH::ptscotch SCOTCH::ptscotcherr)
+    target_compile_definitions(SeisSol-common-properties INTERFACE USE_PTSCOTCH HAVE_GRAPH_PARTITIONING_LIBRARY)
+endif()
+if (NOT GRAPH_PARTITIONING_LIBS)
+    message(WARNING "Compiling without graph partitioning library; the PUML mesh format is disabled.")
```

Comment:
If you do not partition then the distribution of elements to ranks is completely random. Such a build would not be useful in any way for multi-rank runs. Moreover, a great deal of effort has been put into simplifying the installation of SeisSol. There is the spack support and for simple experiments one has the Docker containers.
The build without MPI is imho not useful, too, because no one has tested that for years as far as I know. Therefore, re-enabling no-MPI builds should be another PR that also tests whether everything works.

Reply:
Nevermind, enabled PUML with the "none" partioning lib and added a warning to expect poor performance